client/src/Utils/rfFunctions.ts

```diff
@@ -1,20 +1,40 @@
 export function calculateDate(start: Date, count: number, sampleRate: number) {
-  let time = start.getTime() + count / sampleRate;
+  let time = start.getTime() + (count / sampleRate) * 1e3; // getTime() returns milliseconds
```

Comment:
@Jaydee123 btw this was the only real bug fix, turns out getTime() and Date() operate in ms not s, who knew!